### PR TITLE
MM-11725: Adding new search_autocomplete channels api

### PR DESCRIPTION
**Pull request by jespino**

### CODE REVIEW

File: v4/source/channels.yaml [hunk 1]
```diff
@@ -668,6 +668,45 @@
...
 
+  '/teams/{team_id}/channels/search_autocomplete':
+    get:
+      tags:
+        - channels
+      summary: Autocomplete channels for search
+      description: |
+        Autocomplete your channels on a team based on the search term provided in the request URL.
+
+        __Minimum server version__: 5.3
```
Comment (enahum) [inline v4/source/channels.yaml]: this should be 5.4

**Comment by jespino** [comment]
@enahum PTAL

**Comment by cpanato** [comment]
@jespino this is good to merge?

**Comment by jespino** [comment]
@cpanato Yes

**Pull request merged**